tests/firestore.test.ts

```diff
@@ -1,23 +1,21 @@
 import { expect, test } from '@playwright/test';
 
-
 test('Renders a single document', async ({ page }) => {
 	await page.goto('/firestore-test');
-	await expect( page.getByTestId('doc-data') ).toContainText('Hi Mom');
-
+	await expect(page.getByTestId('doc-data')).toContainText('Hi Mom');
 });
 
 test('Renders a collection of items for an authenticated user in realtime', async ({ page }) => {
 	await page.goto('/firestore-test');
-    
-	await page.click('text=Sign In');
-	await expect( page.getByTestId('count') ).toContainText('0 posts');
-    (await page.waitForSelector('text=Add Post')).click();
-	await expect( page.getByTestId('count') ).toContainText('1 posts');
-	(await page.waitForSelector('text=Add Post')).click();
-	await expect( page.getByTestId('count') ).toContainText('2 posts');
-	expect( (await page.$$('li')).length ).toBe(2);
-	expect(await page.textContent('li')).toContain('firestore item');
-    
+	await page.getByRole('button', { 'name': 'Sign In'}).click();
+	await expect(page.getByTestId('count')).toContainText('0 posts');
+	await page.getByRole('button', { name: 'Add Post' }).click();
+	await expect(page.getByTestId('count')).toContainText('1 posts');
+	await page.getByRole('button', { name: 'Add Post' }).click();
+	await expect(page.getByTestId('count')).toContainText('2 posts');
+	await expect(page.locator('li')).toHaveCount(2);
```

Comment:
General rule, never do manual sync expect's, Playwright has web-first assertions which will retry over and over again until a condition is met or the timeout is reached, Your manual toBe length comparison won't do that. See here: https://playwright.dev/docs/test-assertions